pkg/osvscanner/osvscanner.go

```diff
@@ -87,6 +101,33 @@ func scanDir(r *output.Reporter, query *osv.BatchedQuery, dir string, skipGit bo
 	})
 }
 
+// canCheckIgnore checks if the command `git check-ignore` will work on the system,
+// which is used to check .gitignore files
+func canCheckIgnore() bool {
+	cmd := exec.Command("git", "check-ignore")
```

Comment:
Rather than expecting an external dependency, we should keep our tool as standalone as possible.

Can we instead use https://github.com/src-d/go-git/tree/master/plumbing/format/gitignore ? 

https://cs.github.com/kubernetes-sigs/bom/blob/4415dd6c071a181042f6ebcd197f37608c842f62/pkg/spdx/implementation.go#L499 has an example usage. 

Not sure if there's a higher level convenience parser function, but this appears to be how to read arbitrary .gitignore files which is still quite simple:

https://github.com/src-d/go-git/blob/8b0c2116cea2bbcc8d0075e762b887200a1898e1/plumbing/format/gitignore/dir.go#LL48C14-L48C14 may also provide a higher level/more convenient way to do what we want.

Reply:
Will try using that package.

I did notice that the `git` command was used elsewhere already:
https://github.com/google/osv-scanner/blob/main/pkg/osvscanner/osvscanner.go#L173-L174

maybe we could also replace with go-git there?

Reply:
Good callout! Yes, we should as well.